docs/DeveloperGuide.md

```diff
@@ -1,34 +1,663 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+## How to Use This Guide
 
-## Non-Functional Requirements
+> Icons & labels used in this guide:<br>
+> <br>
+> 💡 - Indicates a tip that may be useful to you.<br>
+> ⚠ - indicates a warning that you should take note of.
+> 
+> _Italic_ - Indicates that the content is a component.<br>
+> `Inline code` - Indicates that the content is a class, method or input by the user.
 
-{Give non-functional requirements}
+&nbsp;
 
-## Glossary
+[🡅 Back to Table of Contents](#table-of-contents)
 
-* *glossary item* - Definition
+----
 
-## Instructions for manual testing
+## Setting up GULIO on Your Computer
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+1. Fork [this repo](https://github.com/AY2021S2-CS2113T-W09-3/tp), and clone the fork into your computer.
+1. Ensure you are using Java 11 or above.
+1. Import the project in your IDE.
+
+> 💡 You are highly recommended to use Intellij IDEA.
+>
+> To set up Intellij:
+> 1. Ensure Intellij is configured to use JDK 11.
+> 1. Import the project as a Gradle project.
+> 1. Verify the setup:
+     >   1. Run `seedu.duke.Duke` and try a few commands.
+>   1. Run the tests to ensure they all pass.
+
+&nbsp;
+
+[🡅 Back to Table of Contents](#table-of-contents)
+
+----
+
+## Introduction
+
+### Background of GULIO
+
+GULIO is a command line application for NUS SOC students to organize their modules. It has a 2-layer system: dashboard layer and module layer. In both layers, the user has access to a different set of commands.
+
+On start up, the user will be on the dashboard layer and has an overview of all their modules. They will have access to module management commands such as adding, deleting or opening a particular module. Opening a module then puts them on the module layer where they can interact with the data within the module.
+
+<p align="center">
+    <img width="973" src="userGuideImages/2-layer.jpg" alt="2-Layer System"><br>
+    Figure 1 - Visualisation of GULIO’s 2-layer system
+</p>
+
+Currently, GULIO is a basic university module manager intended to provide students with an overview and consolidated workspace for all of their modules, lessons, tasks and cheat-sheets. Going forward, we feel that GULIO has the potential for many more features to be added, some of which are proposed in the [Implementation section](#implementation).
+
+### Purpose
+
+This document describes the architecture and implementation details of the command line application, GULIO.
+
+### Scope
+
+This describes the software architecture and software design decisions for the implementation of GULIO. The intended audience of this document is the developers, designers, and software testers of GULIO.
```

Comment:
The `scope` has some duplicated information from `purpose`, which can be redundant.